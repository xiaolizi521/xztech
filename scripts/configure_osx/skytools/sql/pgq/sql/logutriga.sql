
create or replace function pgq.insert_event(que text, ev_type text, ev_data text, x1 text, x2 text, x3 text, x4 text)
returns bigint as $$
begin
    raise notice 'insert_event(%, %, %, %)', que, ev_type, ev_data, x1;
    return 1;
end;
$$ language plpgsql;

create table udata (
    id serial primary key,
    txt text,
    bin bytea
);

create trigger utest AFTER insert or update or delete ON udata
for each row execute procedure pgq.logutriga('udata_que');

insert into udata (txt) values ('text1');
insert into udata (bin) values (E'bi\tn\\000bin');


