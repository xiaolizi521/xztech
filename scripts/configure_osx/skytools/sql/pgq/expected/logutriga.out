create or replace function pgq.insert_event(que text, ev_type text, ev_data text, x1 text, x2 text, x3 text, x4 text)
returns bigint as $$
begin
    raise notice 'insert_event(%, %, %, %)', que, ev_type, ev_data, x1;
    return 1;
end;
$$ language plpgsql;
create table udata (
    id serial primary key,
    txt text,
    bin bytea
);
NOTICE:  CREATE TABLE will create implicit sequence "udata_id_seq" for serial column "udata.id"
NOTICE:  CREATE TABLE / PRIMARY KEY will create implicit index "udata_pkey" for table "udata"
create trigger utest AFTER insert or update or delete ON udata
for each row execute procedure pgq.logutriga('udata_que');
insert into udata (txt) values ('text1');
NOTICE:  insert_event(udata_que, I:id, id=1&txt=text1&bin, public.udata)
CONTEXT:  SQL statement "select pgq.insert_event($1, $2, $3, $4, $5, null, null)"
insert into udata (bin) values (E'bi\tn\\000bin');
NOTICE:  insert_event(udata_que, I:id, id=2&txt&bin=bi%5c011n%5c000bin, public.udata)
CONTEXT:  SQL statement "select pgq.insert_event($1, $2, $3, $4, $5, null, null)"
